Core/Src/internalResistance.c

```diff
@@ -0,0 +1,221 @@
+    // This threshold should always exceed 0, so if avg buffer is saturated with -1000.0f, operation will be skipped
+    if(fabs(deltaCurrent) >= IR_CALC_MIN_CURRENT_DELTA)
+    {
+        // Cycle through every BMB and every brick in the accumulator
+        for(int32_t i = 0; i < NUM_BMBS_PER_PACK; i++)
+        {
+            for(int32_t j = 0; j < NUM_BRICKS_PER_BMB; j++)
+            {
+                // If average current buffer containes placeholder value, -1000.0f, it is excluded from this calculation cycle
+                if((voltageAvgBuffer[i][j][maxCurrentIndex] > -900.0f) && (voltageAvgBuffer[i][j][minCurrentIndex] > -900.0f))
+                {
+                    float deltaVoltage = voltageAvgBuffer[i][j][maxCurrentIndex] - voltageAvgBuffer[i][j][minCurrentIndex];
+                    bms->bmb[i].brickResistance[j] = deltaVoltage / deltaCurrent;
+                }
+            }    
+        }
+    }
+
+    // Static data ready vars set to 0 for next discrete buffer filling cycle 
+    currentDataReady = false;
+    voltageDataReady = false;
+
+    // Reset discrete buffer pointers so data begins filling from index 0 
+    discreteCurrentBufferIndex = 0;
+    discreteVoltageBufferIndex = 0;
+}
+
+/* ==================================================================== */
+/* =================== GLOBAL FUNCTION DEFINITIONS ==================== */
+/* ==================================================================== */
+
+void putCurrentBuffer(Bms_S* bms)
+{   
+    // If any data in the discrete buffer is from a faulty sensor, set that the current data is bad
+    if(bms->currentSensorStatusLO == SENSE_GOOD)
+    {
+        // Update discrete data buffers with current bms current data
+        currentDiscreteBuffer[discreteCurrentBufferIndex] = (bms->tractiveSystemCurrent / 25.0f);
+    }
+    else
+    {
+        currentDataGood = false;
+    }
+
+    // Cycle buffer and wrap to 0 if needed
+    if(++discreteCurrentBufferIndex >= DISCRETE_CURRENT_BUFFER_SIZE)
+    {
+        discreteCurrentBufferIndex = 0;
+
+        // Indicate that the discrete current buffer has been filled
+        currentDataReady = true;
+
+        // Perform a IR calculation if the voltage data is also ready
+        if(voltageDataReady)
+        {
+            calculateInternalResistance(bms);
+        }
+    }
+}
+
+void putVoltageBuffer(Bms_S* bms)
+{
+    // Update discrete data buffers with current bms voltage data
+    for(int32_t i = 0; i < NUM_BMBS_PER_PACK; i++)
+    {
+        for(int32_t j = 0; j < NUM_BRICKS_PER_BMB; j++)
+        {
+            // If any data in the discrete buffer is from a faulty sensor, set that the voltage data is bad
+            if(bms->bmb[i].brickVStatus[j] == GOOD)
+            {
+                voltageDiscreteBuffer[i][j][discreteVoltageBufferIndex] = bms->bmb[i].brickV[j];
+            }
+            else
+            {
+                voltageDataGood[i][j] = false;
```

Comment:
why loop through here and set it to false 84 times? We should set it false once and then `break;`